fe/fe-core/src/main/java/com/starrocks/sql/optimizer/task/SeriallyTaskScheduler.java

```diff
@@ -62,12 +63,39 @@ public void executeTasks(TaskContext context) {
             }
             OptimizerTask task = tasks.pop();
             context.getOptimizerContext().setTaskContext(context);
-            task.execute();
+            if (context.getOptimizerContext().getTraceInfo().isVerboseOptimizer()) {
+                executeWithRecord(task);
+            } else {
+                task.execute();
+            }
         }
     }
 
     @Override
     public void pushTask(OptimizerTask task) {
         tasks.push(task);
     }
+
+
+    private void executeWithRecord(OptimizerTask task) {
+        String timerName = "";
+        if (task instanceof ApplyRuleTask) {
+            timerName = "Optimizer.CostBaseOptimize.ApplyRuleTask";
+        } else if (task instanceof DeriveStatsTask) {
+            timerName = "Optimizer.CostBaseOptimize.DeriveStatsTask";
+        } else if (task instanceof EnforceAndCostTask) {
+            timerName = "Optimizer.CostBaseOptimize.EnforceAndCostTask";
+        } else if (task instanceof ExploreGroupTask) {
+            timerName = "Optimizer.CostBaseOptimize.ExploreGroupTask";
+        } else if (task instanceof OptimizeExpressionTask) {
+            timerName = "Optimizer.CostBaseOptimize.OptimizeExpressionTask";
+        } else if (task instanceof OptimizeGroupTask) {
+            timerName = "Optimizer.CostBaseOptimize.OptimizeGroupTask";
+        } else if (task instanceof RewriteTreeTask) {
+            timerName = "Optimizer.RuleBaseOptimize.RewriteTreeTask";
+        }
+        try (PlannerProfile.ScopedTimer ignore = PlannerProfile.getScopedTimer(timerName)) {
```

Comment:
"Optimizer.CostBaseOptimize." + task.getClass().getSimpleName()

Reply:
Good advice. We need the prefix to represen the hierarchy.

Reply:
the hierarchy don't depend on the prefix

Reply:
Oh，you are right. The level is used for controlling Indentation. But the comment `It's worth to note that, "." in `ComponentA.CompoenentB.FunctionC` is preserved for representing hierarchy` is a little confused.